### ISSUE TITLE
[Suggestion] Offer to delete asociated images when removing keyframe

### ISSUE THREAD

**Issue by degriz** [buddhi1980/mandelbulber2#882]
What it says..

**Comment by buddhi1980** [comment]
Which frames should be deleted? From deleted keyframe to the end? Deleting keyframe affects all keyframes after deleted one.

**Comment by mclarekin** [comment]
i think it is delete rendered frames either side of deleted keyframe to
where the interpolation will not change the image. This is dependent on the
interpolation method ie if  30 frames per keyframe then with linear we only
need to remove 29 images before plus the keyframe image, and 29 after, and
there are no frames to be re-rendered

. Whereas other interpolations may require 3 times this amount of images to
be removed, plus the re-rendering of images affected by interpolation

Basically automating what we have to do manually.

Nice to have but  of low  priority on the to do list

**Comment by degriz** [comment]
Just the frames from that key. Otherwise they get left behind and confuse the count. End up with lumps of prior version.

**Comment by mclarekin** [comment]
I am still confused.  "Just the frames *from* that key".  From (forward?)
or belonging to?

 If you delete a keyframe, it has a ripple effect with cubic interpolation.

So if at  30 frames per keyframe,   and you delete a keyframe, how many
associated frames do you want removed?

**Comment by buddhi1980** [comment]
There is bigger issue. If I delete keyframe and frames after it are already rendered, then frame indexes will no longer correspond to keyframe indexes after deleted keyframe.

**Comment by TasMania17** [comment]
A Mandelbulber animation is a .fract file. The images rendered are stored outside Mandelbulber in the 'path for images' folder. Keyframes can be dropped as long as you 'refresh table'. That rebuilds indexes in the fract file only as far as I know.

If a keyframe is discarded after it's images are rendered then it looks like a file manager operation to delete the frames it produced and all subsequent frames. (Unless you want to get into messy file operations of re-numbering frames and syncing with the fract file). You need to get it right before rendering :)

I've made Mandelbulber animations and if you don't like a series of frames then drop another animation stream over the top of them in your video editing software. I don't think Mandelbulber should have to find 4 different types of image files and their alpha channels because it's a fractal rendering program.
